worksheet_D/oxo.py

```diff
@@ -0,0 +1,99 @@
+class OxoBoard:
+    def __init__(self):
+        self.gameBoard = {(0, 0):0, (0, 1):0, (0, 2):0,                 # Creates dictionary data structure.
+                          (1, 0):0, (1, 1):0, (1, 2):0,                 # Represents the state of the game.
+                          (2, 0):0, (2, 1):0, (2, 2):0}
+
+    def get_square(self, x, y):
+        return  self.gameBoard[x, y]                                    # Returns value of this square.
+
+    def set_square(self, x, y, mark):
+        if self.gameBoard[x, y] == 0:                                   # If square is empty mark it.
+            self.gameBoard[x, y] = mark
+            return True
+        else:
+            return False                                                # If not empty return false.
+
+    def is_board_full(self):
+        for x in xrange(3):                                             # Check through each square in the grid.
+            for y in xrange(3):
+                if self.gameBoard[x, y] == 0:                           # If any square contains 0 then grid not full.
+                    return False
+        return True
+
+    def get_winner(self):
+        for i in xrange(3):                                             # Checks each row, column and diagonal.
+            if self.gameBoard[i, 0] == self.gameBoard[i, 1] == self.gameBoard[i, 2] and self.gameBoard[i, 0] != 0:

```

Comment:
Might be worth commenting which of these checks are for the rows, which for the columns, which for the diagonals